github/pkg/adapter/adapter_test.go

```diff
@@ -402,236 +430,115 @@ var testCases = []testCase{
-		h := &fakeHandler{
-			//handler: tc.sink,
-			handler: sinkAccepted, // No tests expect the sink to do anything interesting
-		}
-		sinkServer := httptest.NewServer(h)
-		defer sinkServer.Close()
+func TestGracefullShutdown(t *testing.T) {
+	ce := adaptertest.NewTestClient()
+	ra := newTestAdapter(t, ce)
+	stopCh := make(chan struct{}, 1)
+
+	go func(stopCh chan struct{}) {
+		defer close(stopCh)
+		time.Sleep(time.Second)
+
+	}(stopCh)
 
-		ra, err := New(sinkServer.URL, testOwnerRepo)
+	t.Logf("starting webhook server")
+	err := ra.Start(stopCh)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestServer(t *testing.T) {
+	for _, tc := range testCases {
+		ce := adaptertest.NewTestClient()
+		adapter := newTestAdapter(t, ce)
+		hook, err := gh.New()
 		if err != nil {
 			t.Fatal(err)
 		}
+		router := adapter.newRouter(hook)
+		server := httptest.NewServer(router)
+		defer server.Close()
 
-		t.Run(tc.name, tc.runner(t, *ra))
+		t.Run(tc.name, tc.runner(t, server.URL, ce))
 	}
 }
 
 // runner returns a testing func that can be passed to t.Run.
-func (tc *testCase) runner(t *testing.T, ra Adapter) func(t *testing.T) {
+func (tc *testCase) runner(t *testing.T, url string, ceClient *adaptertest.TestCloudEventsClient) func(t *testing.T) {
 	return func(t *testing.T) {
 		if tc.eventType == "" {
 			t.Fatal("eventType is required for table tests")
 		}
-		eventID := "12345"
-		if tc.eventID != "" {
-			eventID = tc.eventID
+		body, _ := json.Marshal(tc.payload)
+		req, err := http.NewRequest("POST", url, bytes.NewReader(body))
+		if err != nil {
+			t.Fatal(err)
 		}
-		hdr := http.Header{}
-		hdr.Set("X-GitHub-Event", tc.eventType)
-		hdr.Set("X-GitHub-Delivery", eventID)
-		evtErr := ra.handleEvent(tc.payload, hdr)
 
-		if err := tc.verifyErr(evtErr); err != nil {
+		req.Header.Set(GHHeaderEvent, tc.eventType)
+		req.Header.Set(GHHeaderDelivery, eventID)
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
 			t.Error(err)
 		}
-	}
-}
-
-func (tc *testCase) handleRequest(req *http.Request) (*http.Response, error) {
-
-	codec := cehttp.Codec{}
-
-	body, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		return nil, err
-	}
-	msg := &cehttp.Message{
-		Header: req.Header,
-		Body:   body,
-	}
-
-	event, err := codec.Decode(context.Background(), msg)
-	if err != nil {
-		return nil, fmt.Errorf("unexpected error decoding cloudevent: %s", err)
-	}
-
-	if tc.wantCloudEventType != "" && tc.wantCloudEventType != event.Type() {
-		return nil, fmt.Errorf("want cloud event type %s, got %s",
-			tc.wantCloudEventType, event.Type())
-	}
-
-	gotSource := event.Source()
-	if testSource != gotSource {
-		return nil, fmt.Errorf("want source %s, got %s", testSource, gotSource)
-	}
+		defer resp.Body.Close()
 
-	gotSubject := event.Context.GetSubject()
-	if tc.wantCloudEventSubject != "" && tc.wantCloudEventSubject != gotSubject {
-		return nil, fmt.Errorf("want subject %s, got %s", tc.wantCloudEventSubject, gotSubject)
+		t.Helper()
```

Comment:
This should be inside the `validateAcceptedPayload`, on the very first line of the function.

Reply:
Done.